### ISSUE TITLE
Generic Image 10.0.17763.2686 fails to start SQLBrowser

### ISSUE THREAD

**Issue by marknitek** [microsoft/navcontainerhelper#2375]
**Describe the issue**
When using the newest Generic Image available for our Windows Server Host 10.0.17763.2686, the container creation fails upon starting the SQLBrowser Service.

When i try an older image (in my case 10.0.17763.437) it works fine. That enlines with the fact that it was working fine a few weeks ago.
This is running on Windows  Server 2019 with process isolation, windows defender was uninstalled for testing but did not make any difference.

**Scripts used to create container and cause the issue**

This produces the error:
```
PS C:\WINDOWS\system32> $url =Get-ArtifactUrlFromTag -Tag "onprem/19.2.32968.33504/ch"
PS C:\WINDOWS\system32> New-NavContainer -accept_eula -accept_outdated -containerName test1 -artifactUrl $url -auth NavUserPassword
```

while this works fine:
```
PS C:\WINDOWS\system32> New-NavContainer -accept_eula -accept_outdated -containerName test1 -artifactUrl $url -auth NavUserPassword -useGenericImage mcr.microsoft.com/businesscentral:10.0.17763.437
```

**Full output of scripts**
```
BcContainerHelper is version 3.0.3
BcContainerHelper is running as administrator
Host is Microsoft Windows Server 2019 Standard - ltsc2019
Docker Client Version is 18.09.2
Docker Server Version is 18.09.2
Removing container test1
Removing c:\programdata\navcontainerhelper\Extensions\test1
Fetching all docker images
Fetching all docker volumes
Using image mcr.microsoft.com/businesscentral:10.0.17763.2686
Creating Container test1
Style: onprem
Multitenant: No
Version: 19.2.32968.33504
Platform: 19.0.32956.33475
Generic Tag: 1.0.2.1
Container OS Version: 10.0.17763.2686 (ltsc2019)
Host OS Version: 10.0.17763.316 (ltsc2019)
WARNING: Host OS and Base Image Container OS doesn't match and Hyper-V is not installed. If you encounter issues, you could try to install Hyper-V.
Using process isolation
Using locale de-CH
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Files in c:\programdata\navcontainerhelper\Extensions\test1\my:
- AdditionalOutput.ps1
- MainLoop.ps1
- SetupVariables.ps1
- updatecontainerhosts.ps1
Creating container test1 from image mcr.microsoft.com/businesscentral:10.0.17763.2686
20617728a2f997ee9ed6d3f54a3cf698acaa01d20007d33cfb859c4bddcdd0fe
Waiting for container test1 to be ready
Using artifactUrl https://bcartifacts.azureedge.net/onprem/19.2.32968.33504/ch
Using installer from C:\Run\150-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Service 'SQL Server Browser (SQLBrowser)' cannot be started due to the following error: Cannot start service SQLBrowser on computer '.'.
at <ScriptBlock>, C:\Run\navinstall.ps1: line 62
at <ScriptBlock>, C:\Run\start.ps1: line 225
at <ScriptBlock>, <No file>: line 1Error
Using artifactUrl https://bcartifacts.azureedge.net/onprem/19.2.32968.33504/ch
Using installer from C:\Run\150-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Service 'SQL Server Browser (SQLBrowser)' cannot be started due to the following error: Cannot start service SQLBrowser on computer '.'.
at <ScriptBlock>, C:\Run\navinstall.ps1: line 62
at <ScriptBlock>, C:\Run\start.ps1: line 225
at <ScriptBlock>, <No file>: line 1
New-NavContainer Telemetry Correlation Id: fb31a585-2e26-4654-8781-bb2217edbdf1```
```
**Screenshots**

**Additional context**
- does it happen all the time?
- did it use to work?

**Comment by freddydk** [comment]
This is caused by having an un-updated host using a newer container image - the difference between the old host and the new generic image is simply too large.
While I could consider selecting a closer image - I think the real issue here is the host.
You should really really really consider to update your host - it is a serious risk running with a Windows Server 2019 that old.

**Comment by marknitek** [comment]
@freddydk while i tottally agree with you, it did work for a long time now and i would appreciate if the logic could look for a "closer" image. what i think it did in the past? I mean it worked for years.

Being up to date in the (windows/process isolation) docker world has caused a lot of trouble in the past which unfortunatly leads to update fear... 
But yes updating is on our list too.

**Comment by freddydk** [comment]
Well, the reason is that the newer windows container is no longer compatible - I don't know why.
But... - I don't understand why it doesn't select a closer image - could you try running the following commands:

Get-BestGenericImageName -onlyMatchingBuilds

and 

Get-BestGenericImageName

and let me know what they return

**Comment by marknitek** [comment]
The issue is here:
```
$version = $myversions | Where-Object { $_ -le $hostOsVersion } | Select-Object -Last 1
```
after the above mentioned commit, the fist available version would be .437 which is not -le then .316 that i currently have

In my case a higher build would work, but i understand that you changed the logic to only look for earlier builds. 
But why were so many builds removed? Do they not exist anymore?

For the time beeing i have changed the bccontainerhelper.config to point to the image that is working for me. I guess i have to update the host now(-ish) (many containers to recreate)

**Comment by freddydk** [comment]
Yeah - there is something wrong with the logic.
It was changed to ensure it worked with Windows 11 / Windows Server 2022
BTW - the hardcoded list is only used if the machine fails the download - the list contains the current updated images.

**Comment by freddydk** [comment]
This problem only happens if the host OS is older than any of the existing container OS'
Then the new logic selects the latest image - it shouldn't - it should select the first.

**Comment by freddydk** [comment]
fixed in 3.0.4
